### PR TITLE
Stackup 3d/enhancement/add methods and examples

### PR DESCRIPTION
**Pull request by jeremich41**
Add an methods and example about machine learning algorithm.

### CODE REVIEW

File: pyaedt/modeler/stackup_3d.py [hunk 1]
```diff
@@ -1,9 +1,15 @@
+import os
 from collections import OrderedDict
 
+import joblib
```
Comment (maxcapodi78) [inline pyaedt/modeler/stackup_3d.py:4]:
```suggestion
try:
    import joblib
except ImportError:
    pass
```

File: pyaedt/modeler/stackup_3d.py [hunk 2]
```diff
@@ -2318,3 +2465,420 @@ def points_on_layer(self):
...
+        signal_layer,
+        dielectric_layer,
+        patch_length=None,
+        patch_position_x=0,
+        patch_position_y=0,
+        patch_name="patch",
+        reference_system=None,
+        axis="X",
+    ):
+        CommonObject.__init__(self, application)
```
Comment (maxcapodi78) [inline pyaedt/modeler/stackup_3d.py:2487]:
```suggestion
        try:
            joblib
         except NameError:
             raise ImportError("joblib package is needed to run ML.")
        CommonObject.__init__(self, application)
```

**Review by maxcapodi78 — COMMENTED** [review]
I would make the joblib optional with try import.

File: _unittest/test_16_3d_stackup.py [hunk 3]
```diff
@@ -110,3 +113,11 @@ def test_06_hide_variables(self):
         assert self.st.dielectric_x_position.read_only_variable()
         assert self.st.dielectric_x_position.hide_variable(False)
         assert self.st.dielectric_x_position.read_only_variable(False)
+
+    def test_07_ml_patch(self):
```
Comment (maxcapodi78) [inline _unittest/test_16_3d_stackup.py:117]:
```suggestion
    @pytest.mark.skipif(is_ironpython, reason="pytest is not supported with IronPython.")
    def test_07_ml_patch(self):
```

Comment (maxcapodi78) [reply _unittest/test_16_3d_stackup.py:117]: make sure to add is_ironpython variable in the import of the file

**Review by maxcapodi78 — CHANGES_REQUESTED** [review]
Make sure to apply the suggestion to avoid failures in ironpython

**Review by maxcapodi78 — APPROVED** [review]

File: _unittest/test_16_3d_stackup.py (hunk 3 above)
Comment (jeremich41) [reply _unittest/test_16_3d_stackup.py:117]: ml_patch works even in IronPython, it just create a normal patch with a warning message telling that ml doesn't works in Iron

Comment (maxcapodi78) [reply _unittest/test_16_3d_stackup.py:117]: probably that file is skipped in Ironpython.

**Pull request merged**